### PR TITLE
Array sniffs documentation (Related to #1722)

### PR DESCRIPTION
**Pull request by Mike-Hermans**
Adds documentation for the following sniffs:
- Arrays.MultipleStatementAlignment
- Arrays.ArrayKeySpacingRestrictions

Related to #1722

### CODE REVIEW

File: WordPress/Docs/Arrays/ArrayKeySpacingRestrictionsStandard.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<documentation title="Array Key Spacing Restrictions">
+    <standard>
+    <![CDATA[
+      When referring to array items, only include a space around the index if it is a variable.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Correct whitespace for index keys">
```
Comment (GaryJones) [inline WordPress/Docs/Arrays/ArrayKeySpacingRestrictionsStandard.xml]: Start with `Valid: `.

File: WordPress/Docs/Arrays/ArrayKeySpacingRestrictionsStandard.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <![CDATA[
+$post       = $posts[ $post_id ];
+$post       = $posts[123];
+$post_title = $post['post_title'];
+        ]]>
+        </code>
+        <code title="Incorrect spacing">
```
Comment (GaryJones) [inline WordPress/Docs/Arrays/ArrayKeySpacingRestrictionsStandard.xml]: Start with `Invalid: `.

File: WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<documentation title="Multiple Statement Alignment">
+    <standard>
+    <![CDATA[
+      There should be one space on either side of a double arrow operator used to assign a value to a key. In the case of a block of related assignments, more space may be inserted to promote readability.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Double arrow operators aligned">
```
Comment (GaryJones) [inline WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml]: Start with `Valid: `.

File: WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        <![CDATA[
+$args = array(
+    'cat'      <em>=></em> 22,
+    'year'     <em>=></em> $current_year,
+    'monthnum' <em>=></em> $current_month,
+    'order'    <em>=></em> 'ASC',
+);
+        ]]>
+        </code>
+        <code title="Not aligned; harder to read">
```
Comment (GaryJones) [inline WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml]: Start with `Invalid: `.

**Comment by jrfnl** [comment]
@Mike-Hermans Just wondering if you'll have a chance to finish this off in the near future.
I'd like to release WPCS 2.2.0 soon and it would be great if this PR could be included.

If you haven't got time or lost interest, please let us know and we'll see if we can find someone to take over.

**Comment by Mike-Hermans** [comment]
I've updated the examples in ArrayKeySpacingRestrictions to include both a concatenated string and a constant.

For MultipleStatementAlignment I've added another block with code comparision to show the correct spacing in single line arrays, and the multiple spaces for alignment in multi-line arrays.

File: WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<documentation title="Multiple Statement Alignment">
+    <standard>
+    <![CDATA[
+      When declaring arrays, there should be one space on either side of a double arrow operator used to assign a value to a key.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: correct spacing between the key and value.">
+        <![CDATA[
+$args( 'cat'<em> => </em>22, 'year'<em> => </em>$current_year );
```
Comment (jrfnl) [inline WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml]: Multi-item associative arrays are not allowed to be single-line according to the WP Coding Standards, so while this sniff doesn't address that, the example does not comply with the other coding standards.

It also isn't an array declaration the way it is now (it's actually a parse error).

May I suggest splitting this and having two code samples for both `valid` as well as `invalid` for this block ?

```xml
        <code title="Valid: correct spacing between the key and value.">
        <![CDATA[
$foo = array( 'cat'<em> => </em>22 );
$bar = array( 'year'<em> => </em>$current_year );
        ]]>
        </code>
        <code title="Invalid: No or incorrect spacing between the key and value.">
        <![CDATA[
$foo = array( 'cat'<em>=></em>22 );
$bar = array( 'year'<em>=>   </em>$current_year );
        ]]>
        </code>
```

File: WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        </code>
+        <code title="Invalid: No or incorrect spacing between the key and value.">
+        <![CDATA[
+$args( 'cat'<em>=></em>22, 'year'<em>=> </em>$current_year );
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+      In the case of a block of related assignments, more space may be inserted to promote readability.
```
Comment (jrfnl) [inline WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml]: If you're updating the file anyway, I think making this one a bit less like a _mild suggestion_ would be good idea.

```suggestion
      In the case of a block of related assignments, it is recommended to align the arrows to promote readability.
```

**Pull request merged**